### PR TITLE
Debugging updates

### PR DESCRIPTION
**Pull request by georgediaz88**
To alleviate troubleshooting, I've introduced the gem 'pry' (for development) to this project. I find myself using it daily to debug methods and requests. For instance, you can type:

binding.pry

in an action and when making a request, it will create a breakpoint where you've added that line and you can inspect anything at that moment in time (ie. params, variables etc). This is viewable in your terminal and typing 'exit' will exit and continue with the request. Just remember, to take the binding.pry line out of your code when pushing. 

Also added was better errors which gives a nice UI experience for errors. Used by the Rails community, this provides the same stacktrace with a better look and feel in development. You can also access variables etc wherever the page broke because it uses REPL which gives you an interactive session w/the error.

I've also cleaned up the verbose messages from .rvmrc by updating it to the new standards.

Useful links:
Pry => 
  https://github.com/pry/pry
  http://railscasts.com/episodes/280-pry-with-rails
Better Errors =>
  https://github.com/charliesome/better_errors
  http://railscasts.com/episodes/402-better-errors-railspanel

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -0,0 +1 @@
+ruby-1.9.3-p327
```
Comment (hannahwhy) [inline .ruby-version]: Just FYI: This should be ruby-1.9.3-p429 now.

**Comment by hannahwhy** [comment]
Two questions:
1. What pain points motivated this?
2. What are the effects of these additions on application startup and the rest of the development process?

File: .ruby-version (hunk 1 above)
Comment (georgediaz88) [inline .ruby-version]: I may have had a lesser version due to a known issue via the setup but will up it.

**Comment by georgediaz88** [comment]
1. Understanding what is happening in the models, controllers, and views for this project. Using pry can immensely help in understanding what is happening and what variables are set to. Also, if there are errors on pages, you can pry and fix the errors quicker (I believe) than having to dig deeper into an issue.

Nothing should change on development startup as I didn't change anything in development.rb or wanted to scare anyone when they start a console. Thus, I remember consulting this with you and making sure that part would not be added when adding pry.

If you or anyone would like this to be a default console they can always create an .irbrc , .pryrc file as listed here:
- http://filipemoreira.com/blog/2012/03/09/using-pry-with-rails
  so that code doesn't have to change to reflect this.

**Comment by phebal** [comment]
With `better_errors` is there an option to view the output in plaintext?  Also can you rebase to combine 869a537 and b22c35e, they seem to belong in the same commit. Other then this I think this is fine to merge unless someone objects.

**Comment by georgediaz88** [comment]
@phebal the way to go back would just be to require false on the better_errors gem. The point of that gem is to provide a nice UI navigating feel to what Rails defaults projects with. The main 'cool' thing is to interact with the code on that page which shows you where the error occurred.

**Comment by georgediaz88** [comment]
Updated. Combined gem additions to one commit and corrected ruby version to use.

**Comment by jefflunt** [comment]
I don't have any objections as the `debugger` gem is still in the `Gemfile`. So long as people using the current debugger are unaffected then I'm okay with that. @georgediaz88 - did you confirm that the current `debugger` gem functionality still works?

We can then, later, remove `debugger` **if** people enjoy `pry` and want to universally convert to it.

**Comment by georgediaz88** [comment]
@normalocity - I agree that sounds like a plan and another reason I didn't default the console just yet with pry. I haven't really fiddled w/debugger too much. It still should be fine though. Is there something I can try to make sure I get correct output?

**Comment by phebal** [comment]
@georgediaz88 - I think that it's valuable and even easier to read to have a trace stack in a plain text format but I don't think it's important enough to block this PR since I can still get it from the logs.

**Comment by georgediaz88** [comment]
@phebal it's still there if you look at the left side of this image

https://a248.e.akamai.net/camo.github.com/f05d967fb90cbe3e686ad794062c2151f7ee19a5/687474703a2f2f692e696d6775722e636f6d2f7a594f58462e706e67

**Comment by phebal** [comment]
@georgediaz88 it's there but only in html format and not plaintext.

**Comment by jefflunt** [comment]
@georgediaz88 I would feel best if you sat with someone who uses the `debugger` currently and made sure they've verified that their workflow either isn't affected, or is affected in ways they are okay with (and then document what was found in this ticket - no impact or impact with an explanation).

It's important that we understand what functionality we're replacing/duplicating before just merging this. I'm personally a fan of `pry` as well, but if the `better_errors` obfuscate stack traces for people on the command line or in the console that's going to affect many people.

I think as the person proposing the merge, I'd like to see you do the leg work to justify the change and get buy in from fellow developers.

I think all we need is:
- What's the current debugging use case (this should be understood before replacing it)
- What effect, if any, does `better_errors` have on the way stack traces and errors are displayed in the logs?

Finally, if this is going to take more than 30 minutes and isn't going to save us time in the long run, then we really need to work on more important issues connected to the probation and come back to this later. There's no need to rush this pull request.

**Comment by georgediaz88** [comment]
I'll say this @phebal @normalocity, until I find a way to get a plain text stacktrace, I'll remove better errors until we decide we may find it "Extra" useful to tap in code which is a useful feature it provides. As for pry, I'd really love to keep that as I feel you guys can agree with. So, yes, let me find out if it affects debugger which I believe it won't and  when I find that ou,t I'll happily respond to my findings so we can at least add pry.

**Comment by jefflunt** [comment]
@geogediaz88 - @phebal - I'm okay adding ONLY adding `pry` for now.

**Pull request merged**